xml-occurrence-parsers/src/main/java/org/gbif/xml/occurrence/parser/parsing/xml/XmlFragmentParser.java

```diff
@@ -208,4 +213,34 @@ public static RawOccurrenceRecord parseRecord(byte[] xml, OccurrenceSchemaType s
 
     return results;
   }
+
+  /**
+   * Filters the records by discarding the ones without ID.
+   */
+  private static List<RawOccurrenceRecord> filterRecords(List<RawOccurrenceRecord> records) {
+    if (records == null) {
+      return Collections.emptyList();
+    }
+
+    List<RawOccurrenceRecord> result = new ArrayList<>(records.size());
+    for (RawOccurrenceRecord record : records) {
+      // try to set an id in case it is missing
+      if (Strings.isNullOrEmpty(record.getId())) {
+        // try to create a triplet
+        Triplet triplet = null;
+        try {
+          triplet = new Triplet(record.getInstitutionCode(), record.getCollectionCode(), record.getCatalogueNumber());
+        } catch (Exception e) {
+          LOG.info("Could not create a triplet for record with institution code {}, collection code {} and catalogue "
```

Comment:
in logback you can add a Throwable and it will be handled as the error context and is ignored as argument, this means that you can do:

LOG.info("Could not create a triplet for record with institution code {}, collection code {} and catalogue "
                   + "number {}", record.getInstitutionCode(), record.getCollectionCode(), record.getCatalogueNumber(), e);

See: https://github.com/qos-ch/logback/blob/master/logback-classic/src/main/java/ch/qos/logback/classic/spi/LoggingEvent.java#L127